proximo/proximo.go

```diff
@@ -11,8 +14,21 @@ import (
 	"github.com/uw-labs/substrate"
 )
 
-func dialProximo(broker string, insecure bool) (*grpc.ClientConn, error) {
+type KeepAlive struct {
+	Time    time.Duration
+	Timeout time.Duration
+}
+
+func dialProximo(broker string, insecure bool, ka *KeepAlive) (*grpc.ClientConn, error) {
 	var opts []grpc.DialOption
+
+	if ka != nil {
+		opts = append(opts, grpc.WithKeepaliveParams(keepalive.ClientParameters{
+			Time:    ka.Time,
```

Comment:
It would be nice to have some sanity checks on these? Or is that done in gRPC when initialising the connection?

Reply:
What sort of sanity checks are we talking about, we're already checking it's a valid duration or the `time.ParseDuration` would simply error.  We could enforce a minimum of 1 second for the `time` but since this is an opensource project I would not want to put any restrictions in place that would prevent this library working with someones network requirements.

Reply:
I meant more checking that it's not negative, but my guess would be that gRPC will complain in that case...